system-contracts/contracts/Compressor.sol

```diff
@@ -39,20 +40,24 @@ contract Compressor is ICompressor, ISystemContract {
     ///         * The 2-byte index of the chunk in the dictionary is added to the encoded data.
     /// @dev Currently, the method may be called only from the bootloader because the server is not ready to publish bytecodes
     /// in internal transactions. However, in the future, we will allow everyone to publish compressed bytecodes.
+    /// @dev Read more about the compression: https://github.com/matter-labs/zksync-era/blob/main/docs/guides/advanced/compression.md
     function publishCompressedBytecode(
         bytes calldata _bytecode,
         bytes calldata _rawCompressedData
     ) external payable onlyCallFromBootloader returns (bytes32 bytecodeHash) {
         unchecked {
             (bytes calldata dictionary, bytes calldata encodedData) = _decodeRawBytecode(_rawCompressedData);
 
-            require(dictionary.length % 8 == 0, "Dictionary length should be a multiple of 8");
```

Comment:
Please restore this check.

Imagine `dictionary.length % 8 == 5`, then contract can read out of bound value from the dictionary that may leads to  unrestorable compression state :( 

```
uint64 encodedChunk = dictionary.readUint64(indexOfEncodedChunk);
```

Reply:
How could it be anything but 0? In `_decodeRawBytecode` we read the `dictionaryLen` from the first 2 bytes and multiply that with 8:
```
            // The dictionary length can't be more than 2^16, so it fits into 2 bytes.
            uint256 dictionaryLen = uint256(_rawCompressedData.readUint16(0));
            dictionary = _rawCompressedData[2:2 + dictionaryLen * 8];
            encodedData = _rawCompressedData[2 + dictionaryLen * 8:];
```
So `dictionary.length % 8 == 0 ` is guaranteed.

Reply:
True, lgtm then